### PR TITLE
Sf api integration

### PR DESCRIPTION
**Pull request by saurabhB-Webonise**

### CODE REVIEW

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueDescriptonDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.sjana.fitness.view.feature.home.explore.valuefinder
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.sjana.fitness.R
+class ValueDescriptonDialogFragment : Fragment() {
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueDescriptonDialogFragment.kt:9]: extend with basedialogfragment

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt [hunk 2]
```diff
@@ -70,4 +85,58 @@ class ValueFinderFragment : BaseFragment<ValueFinderViewModel>(), ValueFinderNav
     override fun openNextActivity() {
     }
 
+    class ValueFinderPagerAdapter(valueFinderFragment: ValueFinderFragment) : FragmentStateAdapter(valueFinderFragment) {
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt:88]: create the adapter car separately

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                }else{
+                   completeCardVfBtn.setTextColor(resources.getColor(R.color.darkTxtColor))
+                }
+            }
+
+        })
+
+    }
+
+}
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt:46]: EOL

File: app/src/main/res/layout/value_finder_fragment.xml [hunk 4]
```diff
@@ -37,71 +37,113 @@
...
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/infoVfTxt" />
 
+    <RelativeLayout
+        android:id="@+id/toggleSectionVf"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="#e5e5e5"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/value_finder_fragment.xml:84]: use this from colors.xml

File: app/src/main/res/layout/value_finder_fragment.xml [hunk 5]
```diff
@@ -37,71 +37,113 @@
...
+
+            <ToggleButton
+                android:id="@+id/listVfTxt"
+                android:layout_width="60dp"
+                android:layout_height="25dp"
+                android:background="@drawable/button_toggle_card"
+                android:gravity="center"
+                android:text="@string/list"
+                android:textColor="@color/colorOffWhite"
+                android:textOff="List"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/value_finder_fragment.xml:127]: no hardcoded strings

File: app/src/main/res/layout/value_finder_fragment.xml [hunk 6]
```diff
@@ -37,71 +37,113 @@
...
+            android:layout_below="@+id/layout"
+            android:layout_centerHorizontal="true"
+            android:layout_marginTop="30dp"
+            android:layout_marginBottom="10dp"
+            android:background="@drawable/button_background"
+            android:gravity="center"
+            android:visibility="gone"
+            android:textColor="@color/colorWhite"
+            android:textSize="22sp"
+            tools:text="Complete" />
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/value_finder_fragment.xml:145]: same here

File: app/src/main/res/layout/value_finder_pager_fragment.xml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        app:layout_constraintTop_toBottomOf="@id/pageNoVcTxt" />
+
+    <TextView
+        android:id="@+id/titleVcTxt"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="7dp"
+        android:gravity="center_horizontal"
+        android:letterSpacing="0.05"
+        android:text="Accountability"
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/value_finder_pager_fragment.xml:42]: here as well

File: app/src/main/res/layout/value_finder_pager_fragment.xml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        android:gravity="center"
+        android:textColor="#757575"
+        android:textOff="@string/mark_as_important"
+        android:textOn="@string/important"
+        android:textSize="18sp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/value_finder_pager_fragment.xml:82]: EOL

**Review by tejasAlsi-Webonise — CHANGES_REQUESTED** [review]
check comments

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt (hunk 2 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderFragment.kt:88]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt (hunk 3 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt:46]: done

File: app/src/main/res/layout/value_finder_fragment.xml (hunk 4 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/value_finder_fragment.xml:84]: done

File: app/src/main/res/layout/value_finder_fragment.xml (hunk 5 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/value_finder_fragment.xml:127]: done

File: app/src/main/res/layout/value_finder_fragment.xml (hunk 6 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/value_finder_fragment.xml:145]: done

File: app/src/main/res/layout/value_finder_pager_fragment.xml (hunk 8 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/value_finder_pager_fragment.xml:82]: done

File: app/src/main/res/layout/value_finder_pager_fragment.xml (hunk 7 above)
Comment (saurabhB-Webonise) [reply app/src/main/res/layout/value_finder_pager_fragment.xml:42]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueDescriptonDialogFragment.kt (hunk 1 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueDescriptonDialogFragment.kt:9]: done

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.CompoundButton
+import androidx.fragment.app.Fragment
+import com.sjana.fitness.R
+import com.sjana.fitness.model.ValueCardData
+import kotlinx.android.synthetic.main.value_finder_pager_fragment.*
+
+class ValueFinderPagerFragment(valueCardData: ValueCardData,currentPageNo:Int,totalPage:Int) : Fragment() {
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt:13]: extend with basefragment

**Review by tejasAlsi-Webonise — CHANGES_REQUESTED** [review]
check comments

File: app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt (hunk 9 above)
Comment (saurabhB-Webonise) [reply app/src/main/java/com/sjana/fitness/view/feature/home/explore/valuefinder/ValueFinderPagerFragment.kt]: done

**Review by tejasAlsi-Webonise — APPROVED** [review]

**Pull request merged**